src/scripts/main.js

```diff
@@ -1,3 +1,46 @@
 'use strict';
 
-// write code here
+const listElements = [...document.getElementsByTagName('li')];
+const ul = document.querySelector('ul');
+const schema = ['name', 'position', 'salary', 'age'];
+
+function sortList(list) {
+  list.sort((prev, next) => {
+    const prevSalary = prev.getAttribute('data-salary');
+    const nextSalary = next.getAttribute('data-salary');
+
+    return convertToNumber(nextSalary) - convertToNumber(prevSalary);
+  });
+
+  ul.append(...listElements);
+}
+
+function convertToNumber(string) {
+  return Number(string.split(',').join('').split('$').join(''));
+}
+
+function getEmployees(list) {
+  const listArray = [];
+
+  for (let i = 0; i < listElements.length; i++) {
+    const personInObject = {};
+    const person = list[i];
+
+    for (let j = 0; j < schema.length; j++) {
+      const property = schema[j];
+
+      if (property === 'name') {
+        personInObject[property] = person.innerText;
+      } else {
+        personInObject[property] = person.dataset[property];
+      }
+    }
+
+    listArray.push(personInObject);
+  }
```

Comment:
Блин, я в шоке, и правда в задании есть такое) Просто открытие для меня, потому что в прошлой группе 50 человек этого не заметили, сделали 1 функцию и всем было норм. Окей, тогда давай разбирать. Смотри, как я понял, нам нужно создать простой обьект, где будут 4 поля, заранее нам известные. Так зачем усложнять с этими схемами, вложенными циклами, че-то там сравнивать? Что мешает просто сделать 
```suggestion
  const person = {
    name: *your code*,
    position: *your code*,
    salary: *your code*,
    age: *your code*,
  }
  
  listArray.push(person)
```
? Так ведь намного меньше кода, и самое главное, это хотя бы возможно читать)

Reply:
Это было на случай, если объект будет больше)
Но согласна, в этом случае это лишнее

Reply:
Если обьект будет больше, то в прошлом случае тебе бы тоже пришлось как минимум помкнять схему. А тогда какая разница, менять схему или сам обьект?